aerospike/datadog_checks/aerospike/aerospike.py

```diff
@@ -58,86 +61,123 @@ def parse_namespace(data, namespace, secondary):
-        namespaces = fp.readline().rstrip().split(';')
-        if required_namespaces:
-            return [v for v in namespaces if v in required_namespaces]
+        self.service_check(SERVICE_CHECK_UP, self.OK, tags=self._tags)
+
+    def collect_info(self, command, metric_type, separator=';', required_keys=None, tags=None):
+        entries = self.get_info(command, separator=separator)
+
+        if required_keys:
+            required_data = {}
+            for entry in entries:
+                key, value = entry.split('=', 1)
+                if key in required_keys:
+                    required_data[key] = value
         else:
-            return namespaces
+            required_data = dict(entry.split('=', 1) for entry in entries)
 
-    def _get_connection(self, addr):
-        conn = self.connections.get(addr, None)
+        if metric_type in AEROSPIKE_CAP_MAP:
+            cap = self.instance.get(AEROSPIKE_CAP_CONFIG_KEY_MAP[metric_type], AEROSPIKE_CAP_MAP[metric_type])
+            if len(required_data) > cap:
+                self.log.warn(
+                    'Exceeded cap `{}` for metric type `{}` - please contact support'.format(cap, metric_type)
+                )
+                return
 
-        if conn is None:
-            conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            conn.connect(addr)
-            self.connections[addr] = conn
+        for key, value in iteritems(required_data):
+            self.send(metric_type, key, value, tags)
 
-        return conn
+    def get_namespaces(self):
+        namespaces = self.get_info('namespaces')
 
-    def _process_throughput(self, data, metric_type, namespaces, tags={}):
-        while data != []:
+        if self._required_namespaces:
+            return [ns for ns in namespaces if ns in self._required_namespaces]
+
+        return namespaces
+
+    def get_client(self):
+        try:
+            client = aerospike.client({'hosts': [self._host]}).connect()
+        except Exception as e:
+            self.log.error('Unable to connect to database: {}'.format(e))
+            self.service_check(SERVICE_CHECK_CONNECT, self.CRITICAL, tags=self._tags)
+        else:
+            self.service_check(SERVICE_CHECK_CONNECT, self.OK, tags=self._tags)
+            return client
+
+    def get_info(self, command, separator=';'):
+        # command\tKEY=VALUE;KEY=VALUE;...
+        data = self._client.info_node(command, self._host)
+
+        # Get rid of command and whitespace
+        data = data[len(command):].strip()
+
+        if not separator:
+            return data
+
+        return data.split(separator)
+
+    def collect_throughput(self, namespaces):
+        data = self.get_info('throughput:')
+
+        while data:
             line = data.pop(0)
 
             # skip errors
             while line.startswith('error-'):
-                if data == []:
+                if not data:
                     return
                 line = data.pop(0)
 
```

Comment:
the comments below this are very enigmatic

Reply:
Yeah no idea 😅

Reply:
we should probably remove comments like that